src/System.Reflection.Metadata/tests/PortableExecutable/DebugDirectoryTests.cs

```diff
@@ -153,10 +153,12 @@ private void ValidateEmbeddedPortablePdb(PEReader reader)
             Assert.Equal(DebugDirectoryEntryType.Reproducible, entries[1].Type);
             Assert.Equal(DebugDirectoryEntryType.EmbeddedPortablePdb, entries[2].Type);
 
-            var provider = reader.ReadEmbeddedPortablePdbDebugDirectoryData(entries[2]);
-            var pdbReader = provider.GetMetadataReader();
-            var document = pdbReader.GetDocument(pdbReader.Documents.First());
-            Assert.Equal(@"C:\Documents.cs", pdbReader.GetString(document.Name));
+            using (var provider = reader.ReadEmbeddedPortablePdbDebugDirectoryData(entries[2]))
+            {
+                var pdbReader = provider.GetMetadataReader();
```

Comment:
The memory is not necessarily managed. Common case is actually a memory-map. Or it could be a blob allocated on a native heap. The MetadataReader is just a decoder. Similarly to BlobReader.